cost_map_core/include/cost_map_core/operators/inflation.hpp

```diff
@@ -0,0 +1,196 @@
+/**
+ * @file /cost_map_core/include/cost_map_core/operators/inflation.hpp
+ */
+/*****************************************************************************
+** Ifdefs
+*****************************************************************************/
+
+#ifndef cost_map_core_INFLATION_HPP_
+#define cost_map_core_INFLATION_HPP_
+
+/*****************************************************************************
+** Includes
+*****************************************************************************/
+
+#include "../CostMap.hpp"
+#include <queue>
+
+/*****************************************************************************
+** Namespaces
+*****************************************************************************/
+
+namespace cost_map {
+
+/*****************************************************************************
+** Inflation Function
+*****************************************************************************/
+
+/**
+ * @brief Function which can compute costs for the inflation layer.
+ *
+ * Inherit from this to generate your own inflation
+ * functions.
+ */
+class InflationComputer {
+public:
+  InflationComputer() {};
+  virtual ~InflationComputer() {};
+
+  /** @brief  Given a distance, compute a cost.
+   *
+   * @param  distance The distance from an obstacle in cells
+   * @return A cost value for the distance
+   **/
+  virtual unsigned char operator()(const float &distance) const = 0;
+};
+
+/**
+ * @brief Function which can compute costs for the inflation layer.
+ *
+ * This class provides a default inflation function which works like the
+ * ROS inflation layer. Inherit from this to generate your own inflation
+ * functions.
+ */
+class ROSInflationComputer : public InflationComputer {
+public:
+  ROSInflationComputer(const float& inscribed_radius, const float& weight);
+
+  virtual ~ROSInflationComputer() {};
+
+  /** @brief  Given a distance, compute a cost.
+   *
+   * @param  distance The metric distance from an obstacle (distance = cell_distance*resolution)
+   * @return A cost value for the distance
+   **/
+  virtual unsigned char operator()(const float &distance) const;
+private:
+  float inscribed_radius_, weight_;
+};
+
+/*****************************************************************************
+** Inflation
+*****************************************************************************/
+
+class Inflate {
```

Comment:
(y) thanks for the explanation (eventhough I don't yet understand everything :smile: )!